indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/ec2/EC2AutoScaler.java

```diff
@@ -108,31 +108,36 @@ public AutoScalingData provision()
         }
       }
 
-      final RunInstancesResult result = amazonEC2Client.runInstances(
-          new RunInstancesRequest(
-              workerConfig.getAmiId(),
-              workerConfig.getMinInstances(),
-              workerConfig.getMaxInstances()
+      RunInstancesRequest request = new RunInstancesRequest(
+          workerConfig.getAmiId(),
+          workerConfig.getMinInstances(),
+          workerConfig.getMaxInstances()
+      )
+          .withInstanceType(workerConfig.getInstanceType())
+          .withPlacement(new Placement(envConfig.getAvailabilityZone()))
+          .withKeyName(workerConfig.getKeyName())
+          .withIamInstanceProfile(
+              workerConfig.getIamProfile() == null
+              ? null
+              : workerConfig.getIamProfile().toIamInstanceProfileSpecification()
           )
-              .withInstanceType(workerConfig.getInstanceType())
-              .withSecurityGroupIds(workerConfig.getSecurityGroupIds())
-              .withPlacement(new Placement(envConfig.getAvailabilityZone()))
-              .withKeyName(workerConfig.getKeyName())
-              .withSubnetId(workerConfig.getSubnetId())
-              .withIamInstanceProfile(
-                  workerConfig.getIamProfile() == null
-                  ? null
-                  : workerConfig.getIamProfile().toIamInstanceProfileSpecification()
-              )
-              .withNetworkInterfaces(
-                  workerConfig.getAssociatePublicIpAddress() == null
-                  ? null
-                  : new InstanceNetworkInterfaceSpecification().withAssociatePublicIpAddress(
-                      workerConfig.getAssociatePublicIpAddress()
-                  )
-              )
-              .withUserData(userDataBase64)
-      );
+          .withUserData(userDataBase64);
+
+      if (workerConfig.getAssociatePublicIpAddress() != null) {
+        request.withNetworkInterfaces(
+            new InstanceNetworkInterfaceSpecification()
+                .withAssociatePublicIpAddress(workerConfig.getAssociatePublicIpAddress())
+                .withSubnetId(workerConfig.getSubnetId())
+                .withGroups(workerConfig.getSecurityGroupIds())
+                .withDeviceIndex(0)
```

Comment:
Instances are not guaranteed to only have one network interface, but support for such dynamic configuration would warrant a larger scale discussion, so leaving it as a common use case of One seems fine.